lib/x11test.pm

```diff
@@ -717,7 +717,7 @@ sub exit_firefox_common {
     # Exit
     send_key 'ctrl-q';
     wait_still_screen 1, 2;
-    send_key_until_needlematch([qw(firefox-save-and-quit xterm-left-open xterm-without-focus)], "alt-f4", 3, 30);
+    send_key_until_needlematch([qw(firefox-save-and-quit xterm-left-open xterm-without-focus)], "alt-f4", 6, 30);
```

Comment:
@michaelgrifalconi 
Seems a timing or performance issue here, I am wondering if we can add some timeout here
`wait_still_screen 1, 2;`

BTW, I would suggest do more VRs on other SLE versions which hit the same issue as well :)

Reply:
Hello, started more validation runs.

I don't think a wait_still_screen at the beginning can help much, the issue is between multiple runs of send_key_until_needlematch.
It will send a alt-f4 key initially, wait for it to close firefox and if firefox is not fast enough to close, send another alt-f4 and then both firefox and xterm close and it fails. Increasing the time between sending these alt-f4 keys is the only relevant factor IMHO

Reply:
Well, the reason why I add the comments is that, I try to do some manual tests on my local setup.
`ctrl-q` key can close the firfox page, then I don't need to send `alt-f4` to force closing it.

Anyway, I agree with you, your changes make more sense!

Reply:
Right! the ctrl-q is the one supposed to close it, I missed that sorry! Maybe giving more time could help! I will try to increase it as well.